R/calcShareETP.R

```diff
@@ -75,20 +73,20 @@ calcShareETP <- function(subtype = c("enduse", "carrier"), feOnly = FALSE) {
   etpFilter <- etp %>%
     as.quitte() %>%
     filter(.data[["period"]] %in% periods,
-           .data[["scenario"]] %in% scen) %>%
-    filter(.data[["variable"]] %in% names(reval),
+           .data[["scenario"]] %in% scen,
+           .data[["variable"]] %in% names(reval),
            !is.na(.data[["value"]])) %>%
     mutate(variable = droplevels(revalue(.data[["variable"]], reval)))
-
-  names(etpFilter)[names(etpFilter) == "variable"] <- shareOf
-
+  colnames(etpFilter)[colnames(etpFilter) == "variable"] <- shareOf
 
   # Extrapolate 'biotrad' share from 'biomod' values for carrier separation
   if (subtype == "carrier") {
     etpFilter <- etpFilter %>%
-      rename(variable = "carrier") %>%
-      toolSplitBiomass(gdppop, varName = "biomod") %>%
-      rename(carrier = "variable")
+      as.quitte() %>%
+      as.magpie() %>%
+      toolSplitBiomass(gdppop, "biomod", dim = "carrier") %>%
+      as.quitte() %>%
```

Comment:
Maybe we could also enable ```toolSplitBiomass``` to handle both quitte and magpie obejcts? Just an idea and definitely does not have to be now.